cognite/neat/rules/models/_rules/information_rules.py

```diff
@@ -1,20 +1,108 @@
+import re
+import warnings
 from datetime import datetime
-from typing import ClassVar
+from enum import StrEnum
+from typing import Any, ClassVar
 
+from pydantic import Field, HttpUrl, TypeAdapter, field_validator
 from rdflib import Namespace
 
-from .base import Prefix, RoleTypes, RuleModel
-from .domain_rules import DomainMetadata
+from cognite.neat.rules import exceptions
+from cognite.neat.rules.models.rdfpath import (
+    AllReferences,
+    Hop,
+    RawLookup,
+    SingleProperty,
+    SPARQLQuery,
+    TransformationRuleType,
+    Traversal,
+)
 
+from .base import (
+    Prefix,
+    RoleTypes,
+    RuleModel,
+    SheetList,
+    prefix_compliance_regex,
+    version_compliance_regex,
+)
+from .domain_rules import DomainClass, DomainMetadata, DomainProperty
 
-class InformationArchitectMetadata(DomainMetadata):
+
+class MatchType(StrEnum):
+    exact = "exact"
+    partial = "partial"
+
+
+class InformationMetadata(DomainMetadata):
     role: ClassVar[RoleTypes] = RoleTypes.information_architect
-    prefix: Prefix
-    namespace: Namespace
-    version: str
-    contributor: str | list[str]
-    created: datetime
-    updated: datetime
+    prefix: Prefix = Field(
+        description="This is used as prefix for generation of RDF OWL/SHACL data model representation",
+    )
+    namespace: Namespace = Field(
+        description="This is used as RDF namespace for generation of RDF OWL/SHACL data model representation "
+        "and/or for generation of RDF graphs.",
+        min_length=1,
+        max_length=2048,
+    )
+
+    name: str = Field(
+        alias="title",
+        description="Human readable name of the data model",
+        min_length=1,
+        max_length=255,
+    )
+
+    version: str | None = Field(
+        description="Data model version",
+        min_length=1,
+        max_length=43,
+    )
+
+    contributor: str | list[str] = Field(
+        description=(
+            "List of contributors to the data model creation, "
+            "typically information architects are considered as contributors."
+        ),
+    )
+
+    created: datetime = Field(
+        description=("Date of the data model creation"),
+    )
+
+    updated: datetime = Field(
+        description=("Date of the data model update"),
+    )
+
+    @field_validator("version")
+    def is_version_compliant(cls, value):
+        if not re.match(version_compliance_regex, value):
+            raise exceptions.VersionRegexViolation(value, version_compliance_regex).to_pydantic_custom_error()
+        else:
+            return value
```

Comment:
Just my personal preference but I find it more readable

If this -> raise exception.


```suggestion
        if not re.match(version_compliance_regex, value):
            raise exceptions.VersionRegexViolation(value, version_compliance_regex).to_pydantic_custom_error()
        return value
```